### PR TITLE
Collection of minor improvements

### PR DESCRIPTION
**Pull request by cbergqvist**
Here are some minor improvements I've made since I started using your shell.nix. Let me know which ones you like/don't. :)

Have some larger commits that I'm holding back for now.

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -60,7 +60,7 @@ pkgs.mkShell {
       echo "Bitcoin Core build nix-shell"
       echo ""
 
-      BCC_EGG=${pkgs.linuxPackages.bcc}/lib/python3.11/site-packages/bcc-0.29.1-py3.11.egg
+      BCC_EGG=${pkgs.linuxPackages.bcc}/${pkgs.python3.sitePackages}/bcc-${pkgs.linuxPackages.bcc.version}-py3.${pkgs.python3.sourceVersion.minor}.egg
```
Comment (0xB10C) [inline shell.nix:76]: Awesome! I ran into this recently again and had to update it manually.

File: shell.nix [hunk 2]
```diff
@@ -29,9 +29,10 @@ pkgs.mkShell {
       # only needed for older versions
       # openssl
 
-      # functional tests
+      # functional tests & linting
       python3
       python3Packages.flake8
+      python3Packages.lief
```
Comment (0xB10C) [inline shell.nix:47]: I didn't know the python lief was usd/required. Do you happen to know where it's being used and for what?

File: shell.nix [hunk 3]
```diff
@@ -14,7 +14,7 @@ pkgs.mkShell {
       libevent
       zeromq
       sqlite
-      db48
+      db48 # Berkeley DB 4.8
```
Comment (0xB10C) [inline shell.nix:19]: we might also want have a way to enable/disable this at some point

File: shell.nix [hunk 4]
```diff
@@ -3,8 +3,10 @@
 #pkgs.clangStdenv.mkDerivation {
 #  name = "libcxxStdenv";
 # clang_13
-
-pkgs.mkShell {
+let
+  inherit (pkgs.lib) strings;
+  jobs = if (strings.hasSuffix "linux" builtins.currentSystem) then "$(($(nproc)))" else "6";
```
Comment (0xB10C) [inline shell.nix:8]: 6 is just a magic value I used to leave a bit of room for other things on my 8-thread CPU while compiling Core. I don't have a strong preference to keep it - not sure what a sensible replacement would be

**Review by 0xB10C — APPROVED** [review]
Looks good! Thanks for breathing a bit of life back into this.

All changes look reasonable. I left a few comments. Will merge soon.

**Comment by 0xB10C** [comment]
My plan is to rebase this on https://github.com/0xB10C/nix-bitcoin-core/pull/6, let the CI run, merge on green and then merge https://github.com/0xB10C/nix-bitcoin-core/pull/6 later (feel free to have a look at it if you want @cbergqvist ).

**Comment by 0xB10C** [comment]
CI is running here: https://github.com/0xB10C/nix-bitcoin-core/actions/runs/9613152065

File: shell.nix (hunk 2 above)
Comment (cbergqvist) [reply shell.nix:47]: Ah, thought I got it into the commit message but it's not there.. `lief` is required by `/test/lint/lint-python.py`.

File: shell.nix (hunk 4 above)
Comment (cbergqvist) [reply shell.nix:8]: Good to know. Yeah, I think one has to use something else than `nproc` on Mac for example. For Linux we could go with something like `nproc-2` combined with some `max()`-expression or whatever nix/(bash?) provides.

File: shell.nix (hunk 3 above)
Comment (cbergqvist) [reply shell.nix:19]: Agreed, what do you think of my approach of enabling/disabling things in cf1e4393178222ee45a35d7a8c514225ab156034?

File: shell.nix (hunk 4 above)
Comment (cbergqvist) [reply shell.nix:8]: For Mac we can copy this from @itornaza (thanks MIT license): https://github.com/itornaza/btc-dot/blob/master/bitcoin#L41

Comment (cbergqvist) [reply shell.nix:8]: Tentative solution: 0e8c532780376ebcb47f548889637e7094f75d76
User specifies `nix-shell --arg spareCores 2` in your case. No `max(cores-2, 1)` logic.

File: shell.nix (hunk 3 above)
Comment (0xB10C) [reply shell.nix:19]: TIL `nix-shell --arg`! I like it.

Comment (cbergqvist) [reply shell.nix:19]: Now covered in #7.

File: shell.nix (hunk 4 above)
Comment (cbergqvist) [reply shell.nix:8]: Now covered in #7.

**Pull request merged**